### ISSUE TITLE
E1136: Value is unsubscritable - incorrect behaviour for custom dict objects.

### ISSUE THREAD

**Issue by ylathouris** [PyCQA/pylint#944]
### Steps to reproduce

As far as I know, the recommended way to create a dictionary like object is to inherit from one of the  Abstract Base Classes provided by the collections module. For example:

```python
import collections

class MyDictLikeObject(collections.MutableSequence):
    """A custom dictionary-like object."""

    def __getitem__(self, key):
        # Get value for the given key (i.e. obj[key])

    def __setitem__(self, key, value):
        # Set value for the given key (i.e. obj[key] = value)

    def __delitem__(self, key):
        # Delete item with the given key (i.e. del obj[key])

    def __iter__(self):
        # Iteration behaviour (i.e. for key in obj)

    def __len__(self, key):
        # Get number of items (i.e.  len(obj)) 
```

This allows you to use a custom object as you would an ordinary `dict` object. For example:

```python
obj = MyDictLikeObject()
obj["foo"] = "bar"
```

### Current Behaviour

Pylint raises an error for the second line of this code snippet - E1136 Value "obj" is unsubscritable. This is incorrect.  

### pylint --version output

* pylint 1.5.4 
* astroid 1.4.4

**Comment by PCManticore** [comment]
I cannot reproduce this using pylint 1.5.6 and astroid 1.4.5, as well as the master version. Can you try with these versions and confirm that it works for you as well?

**Comment by AWhetter** [comment]
Yep we can't reproduce with master on the original test case.
